subsys/bluetooth/shell/csis.c

```diff
@@ -105,20 +105,114 @@ static int cmd_csis_register(const struct shell *sh, size_t argc, char **argv)
 	return 0;
 }
 
+static uint8_t rsi[BT_CSIS_RSI_SIZE];
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+	BT_CSIS_DATA_RSI(rsi),
+};
+
+static int csis_adv_update_rsi(const struct bt_csis *csis, struct bt_le_ext_adv *adv)
+{
+	int err;
+
+	err = bt_csis_generate_rsi(csis, rsi);
+	if (err != 0) {
+		shell_error(ctx_shell, "Failed to generate RSI (err %d)", err);
+		return err;
+	}
+
+	err = bt_le_ext_adv_set_data(adv, ad, ARRAY_SIZE(ad), NULL, 0);
+	if (err != 0) {
+		shell_error(ctx_shell, "Failed to set advertising data (err %d)", err);
+		return err;
+	}
+
+	return 0;
+}
+
+#if defined(CONFIG_BT_PRIVACY)
+static bool adv_rpa_expired_cb(struct bt_le_ext_adv *adv)
+{
+	int err;
+
+	err = csis_adv_update_rsi(csis, adv);
+	if (err != 0) {
+		return false;
+	}
```

Comment:
@Thalley Would it be a good idea to instead of having BT_CSIS_ENC_SIRK_SUPPORT option, rely on BT_PRIVACY instead? So if the privacy is enabled, the SIRK will be encrypted, if disabled, it will be plain text.

Reply:
> Given that a higher layer profile may require the SIRK to be encrypted, but without privacy, I vote that we enable `BT_CSIS_ENC_SIRK_SUPPORT=y` by default if `BT_PRIVACY=y`, but do not _require_ it (spec uses "should" and not "shall"), but do not otherwise restrict it.

In that case, no changes are needed in `Kconfig.csis`, as `BT_CSIS_ENC_SIRK_SUPPORT` is enabled by default anyway